api/src/domain/internship.ts

```diff
@@ -35,3 +36,20 @@ export async function getInternshipById(id: string): Promise<InternshipModel> {
     handleServiceError(error);
   }
 }
+
+export async function getCoursesForInternship(
+  internshipId: string,
+): Promise<CourseModel[]> {
+  try {
+    const internship = await prisma.internship.findUnique({
+      where: { id: internshipId },
+      include: { courses: true },
```

Comment:
This returns the full internship while we're only interested in the courses.

I think this should be…

```suggestion
      select: { courses: true },
```